core/src/edu/cornell/gdiac/main/BouncePlatformModel.java

```diff
@@ -17,13 +22,62 @@ public class BouncePlatformModel extends PlatformModel {
    * The vertical force to be applied to the player when the player bounces on the platform while
    * frozen.
    */
-  private float bounceForce;
+  private float BounceCoefficient;
 
-  public float getBounceForce() {
-    return bounceForce;
+  public float getCoefficient() {
+    return BounceCoefficient;
+  }
+  public void setCoefficient(float c){
+    BounceCoefficient=c;
+  }
+  public BouncePlatformModel() {
+    super();
+    region = null;
+    BounceCoefficient=0.0f;
   }
-
   public void initialize(AssetDirectory directory, JsonValue json) {
     // TODO P2
+    // Used red as debug color for bouncy platforms
+    setName(json.name());
+
+    float[] pos = json.get("pos").asFloatArray();
+    float[] size = json.get("size").asFloatArray();
+    setPosition(pos[0], pos[1]);
+    setDimension(size[0], size[1]);
+
+    // Technically, we should do error checking here.
+    // A JSON field might accidentally be missing
+    setBodyType(json.get("bodytype").asString().equals("static") ? BodyDef.BodyType.StaticBody
+        : BodyDef.BodyType.DynamicBody);
+    setDensity(json.get("density").asFloat());
+    setFriction(json.get("friction").asFloat());
+    setRestitution(json.get("restitution").asFloat());
+    // Setting the bounce coefficient
+    float coefficient = json.get("coefficient").asFloat();
+    setCoefficient(coefficient);
+
+    // Reflection is best way to convert name to color
+    Color debugColor;
+    try {
+      String cname = json.get("debugcolor").asString().toUpperCase();
+      Field field = Class.forName("com.badlogic.gdx.graphics.Color").getField(cname);
+      debugColor = new Color((Color) field.get(null));
+    } catch (Exception e) {
+      debugColor = null; // Not defined
+    }
+    int opacity = json.get("debugopacity").asInt();
+    debugColor.mul(opacity / 255.0f);
+    setDebugColor(debugColor);
+
+    // Now get the texture from the AssetManager singleton
+    String key = json.get("texture").asString();
+    TextureRegion texture = new TextureRegion(directory.getEntry(key, Texture.class));
```

Comment:
While you're making changes, can you use `SimpleObjectJsonParser` to parse most of the data for this object?